StudyHub/StudyHub/Controllers/AssignmentTaskController.cs

```diff
@@ -9,23 +11,22 @@ namespace StudyHub.Controllers;
 public class AssignmentTaskController : Controller
 {
     private readonly IAssignmentTaskService _serviceTask;
+    private readonly AssignmentTaskOptionValidator _optionsValidator;
 
-    public AssignmentTaskController(IAssignmentTaskService serviceTask)
+    public AssignmentTaskController(
+        IAssignmentTaskService serviceTask,
+        AssignmentTaskOptionValidator optionsValidator)
     {
         _serviceTask = serviceTask;
+        _optionsValidator = optionsValidator;
     }
 
     [HttpPost]
     public async Task<IActionResult> InsertAssigmentTask(CreateAssignmentTaskDTO dto)
     {
-        try
-        {
-            var result = await _serviceTask.AddTask(dto);
-            return Ok(result);
-        }
-        catch (Exception ex)
-        {
-            return BadRequest(ex.Message);
-        }
+        _optionsValidator.ValidateAndThrow(dto.Options);
```

Comment:
Can't you validate whole **dto** here, including **dto.Options**?
Seems logically for me

Reply:
Still not resolved :)